cdap-data-fabric/src/main/java/co/cask/cdap/spi/metadata/Metadata.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.spi.metadata;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * The metadata for an entity, consisting of tags and properties.
+ */
+public final class Metadata {
+
+  private final Set<ScopedName> tags;
+  private final Map<ScopedName, String> properties;
+
+  public Metadata(Set<ScopedName> tags, Map<ScopedName, String> properties) {
+    this.tags = tags;
```

Comment:
Should we do a a deep unmodifiable copy for the set and map?

Reply:
god point. done.